editor-components/modal/confirmation-modal-1/confirmation.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from "react";
+import ComposerModalClose from "../../../composer-base-components/close/close";
+import { BaseModal, TypeUsableComponentProps } from "../../EditorComponent";
+import styles from "./confirmation.module.scss";
+import ComposerLink from "../../../../custom-hooks/composer-base-components/Link/link";
+import { ComposerIcon } from "../../../composer-base-components/icon/icon";
+
+
+
+
+
+class ConfirmationModal extends BaseModal{
+    constructor(props?: any) {
+        super(props, styles);
+
+        
+
+        this.addProp({
+            type: "image",
+            key: "image",
+            displayer: "image",
+            value: "https://picsum.photos/200"
+        });
+
+        this.addProp({
+            type: "icon",
+            key: "close-icon",
+            displayer: "icon",
+            value: "AiOutlineClose"
+        })
+
+        this.addProp({
+            type: "string",
+            key: "title",
+            displayer: "Title",
+            value: "Hang On a Sec!"
+        })
+
+        this.addProp({
+            type: "string",
+            key: "description",
+            displayer: "Description",
+            value: "Ready to rock and roll? Confirm your choice, and let's make some magic happen!"
+        })
+
+        this.addProp({
+            type: "object",
+            key: "button-1",
+            displayer: "Button",
+            value: [
+                {
+                    type: "string",
+                    key: "button1-text",
```

Comment:
you can just say text for key. also create a type for this. because it is an object